tokenauth/token.go

```diff
@@ -25,10 +26,20 @@ type Claims struct {
 type TokenAuth struct {
 	authService         *authservice.AuthService
 	acceptRokwireTokens bool
+
+	blacklist     []string
+	blacklistLock *sync.RWMutex
+	blacklistSize int
+	blacklistHead int
 }
 
 // CheckToken validates the provided token and returns the token claims
 func (t *TokenAuth) CheckToken(token string, purpose string) (*Claims, error) {
+	for _, badToken := range t.blacklist {
```

Comment:
We may want to traverse the queue in reverse order since the newest elements will be at the end. It is more likely that tokens we are receiving will match the more recently added elements.